assessments/GOLD/earthmover.yaml

```diff
@@ -0,0 +1,176 @@
+    header_rows: 1
+  performanceLevelDescriptors:
+    file: ./seeds/performanceLevelDescriptors.csv
+    header_rows: 1
+  gradeLevelDescriptors:
+    file: ./seeds/gradeLevelDescriptors.csv
+    header_rows: 1
+
+  input:
+    # THIS FILE DOES NOT EXIST; IT MUST BE SUPPLIED BY THE USER!
+    file: ${INPUT_FILE}
+    header_rows: 1
+    # See the accompanying bundle_metadata.json for a list of required columns for this file
+    # Or see data/sample_anonymized_file.csv
+
+
+{% set assessments = {
+  'Social-Emotional':'GOLD_SE',
+  'Physical': 'GOLD_PHY',
+  'Language': 'GOLD_LANG',
+  'Literacy': 'GOLD_LIT',
+  'Spanish Language': 'GOLD_LANG_Spanish',
+  'Spanish Literacy': 'GOLD_LIT_Spanish',
+  'Cognitive': 'GOLD_COG',
+  'Mathematics':'GOLD_MATH'
+}
+%}
+
+{% set school_years = [
+  'Fall 2022/2023'
+]
+%}
+
+{% set reporting_methods = [
+  'Raw Score',
+  'Scale Score',
+  'WHE',
+  'NN',
+  'K-Readiness'
+]
+%}
+
+transformations:
+  input: #this initial transformation node should not be deleted in order for this bundle to be compatible with the student ID xwalking package
+    source: $sources.input
+    operations: []
+
+# The original ingested data comes in a very wide format, with each observation containing all past assessment results for each student. We need to pivot the data longer by (1) creating 
+# separate "dataframes" in which each observation represents an assessment result for all reporting methods per student, (2) union them all together.
+
+{% for school_year in school_years %} 
+{% for assessment in assessments.keys() %}
+  {{assessment| replace(" ", "_") }}_{{school_year| replace(" ", "_")}}:
+    source: $transformations.input
+    operations:
+      - operation: keep_columns
+        columns:
+          - ChildID
+          - {{school_year ~ ' ' ~ assessment ~' Date Finalized'}}
+          {%- for reporting_method in reporting_methods %}
+          - {{ school_year ~ ' ' ~ assessment ~ ' ' ~ reporting_method }}
+          {%- endfor %}
+      - operation: rename_columns
+        columns:  
+          {{school_year ~ ' ' ~ assessment ~' Date Finalized'}}: admin_date
+          {%- for reporting_method in reporting_methods %}
+          {{school_year ~ ' ' ~ assessment ~ ' ' ~reporting_method}}: {{reporting_method}}
+          {%- endfor %}
+      - operation: add_columns
+        columns:
+          school_year: {{school_year}}
+          assessment: {{assessment}}
+          assessment_id: {{assessments.get(assessment)}}
+          namespace: "uri://www.teachingstrategies.com/assessment/gold"
+      - operation: modify_columns
+        columns:
+          admin_date: "{%raw%}{% if admin_date != '' %}{{value}}{% else %}{% if school_year == 'Fall 2022/2023'%}10/1/2022{% endif %}{% endif %}{%endraw%}"
+      - operation: combine_columns
+        columns:
+          - school_year
+          - admin_date
+          - assessment_id
+          - ChildID
+        new_column: student_assessment_id
```

Comment:
Have we confirmed this convo of columns = uniqueness? My assumption is yes, just double checking!

Reply:
I believe so !